lofo/flofo_importance.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import cross_validate
+from tqdm import tqdm_notebook
+import multiprocessing
+import warnings
+from lofo.infer_defaults import infer_model
+
+
+class FLOFOImportance:
+
+    def __init__(self, predict_func, df, features, target,
+                 scoring, n_jobs=None):
+
+        df = df.copy()
+        self.predict_func = predict_func
+        self.df = df
+        self.features = features
+        self.target = target
+        self.scoring = scoring
+        self.n_jobs = n_jobs
+        if self.n_jobs is not None and self.n_jobs > 1:
+            warning_str = "Warning: If your model is multithreaded, please initialise the number \
+                of jobs of LOFO to be equal to 1, otherwise you may experience issues."
+            warnings.warn(warning_str)
+        if df.shape[0] <= 1000:
+            warnings.warn("Small validation set")
+
+        self._bin_features()
+
+    def _bin_features(self):
+        self.bin_df = pd.DataFrame()
+        for feature in self.features:
+            self.bin_df[feature] = (self.df[feature].rank(pct=True)*(10 - 0.001)).astype(int)
+
+    def _get_score(self, df):
+        return self.scoring._sign*self.scoring._score_func(self.predict_func(df[self.features]), self.df[self.target])
```

Comment:
so no scikit-learn scoring here?

Reply:
actually this is function expect this one
https://github.com/scikit-learn/scikit-learn/blob/0e9520b235522146b7395b6d848319d3810aba97/sklearn/model_selection/_validation.py#L585

Reply:
Or maybe we could use 
https://github.com/scikit-learn/scikit-learn/blob/8a4cb4871d07c0ddd966fce0959d8118a8f62c59/sklearn/metrics/scorer.py#L243


```
scorer = check_scoring(estimator=estimator, scoring=scoring)
score = scorer(estimator, X_test, y_test)
```